File: Programmers/Lv0/수조작하기1/Solution.java
package Lv0.수조작하기1;

class Solution {

    private int move(int n, char c) {
        switch (c) {
            case 'w' -> n++;
            case 's' -> n--;
            case 'd' -> n += 10;
            case 'a' -> n -= 10;
        }
        return n;
    }

    public int solution(int n, String control) {
        for (char c : control.toCharArray()) {
            n = move(n, c);
        }

        return n;
    }
}

File: Programmers/Lv0/수조작하기1/solution.py
def move(n: int, ch: str):
    # python 3.10부터 match case가 추가되었음
    if ch == 'w':
        return n + 1
    elif ch == 's':
        return n - 1
    elif ch == 'd':
        return n + 10
    elif ch == 'a':
        return n - 10

    return n


def solution(n, control):
    for c in control:
        n = move(n, c)

    return n